### PR TITLE
update install script/docs for new apt and yum repos

### PR DESCRIPTION
**Pull request by jessfraz**
This is the next step after https://github.com/docker/docker/pull/13680 is merged. It updated the install script to use the new repos

### CODE REVIEW

File: hack/install.sh [hunk 1]
```diff
@@ -210,18 +221,37 @@ do_install() {
 			fi
 			(
 				set -x
-				if [ "https://get.docker.com/" = "$url" ]; then
+				if [ "https://apt.dockerproject.org/" = "$url" ]; then
 					$sh_c "apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 36A1D7869245C8950F966E92D8576A8BA88D21E9"
```
Comment (jfrazelle) [inline hack/install.sh]: im going to make a new key so this will change

File: hack/install.sh [hunk 2]
```diff
@@ -210,18 +221,37 @@ do_install() {
...
+				(
+					set -x
+					$sh_c 'docker version'
+				) || true
+			fi
+			echo_docker_as_nonroot
+			exit 0
+			;;
+		
+		fedora|centos)
```
Comment (jfrazelle) [inline hack/install.sh]: ping @vbatts i have no idea what im  doing lolz

**Comment by calavera** [comment]
Is this ready to merge @jfrazelle? so far it LGTM.

**Comment by icecrime** [comment]
Yep, not an expert here but LGTM. Ping @thaJeztah @moxiegirl regarding docs!

**Comment by jfrazelle** [comment]
I need to test a bit more on different distros, sorry i will let you know :)

**Comment by jfrazelle** [comment]
But docs can def be reviewed, I just am a little overly cautious

**Comment by calavera** [comment]
moved to docs review :stuck_out_tongue_winking_eye:

File: docs/installation/centos.md [hunk 3]
```diff
@@ -44,45 +44,7 @@ reported kernel bugs may have already been fixed on the latest kernel packages
...
-    https://get.docker.com/rpm/1.7.1/centos-7/SRPMS/docker-engine-1.7.1-1.el7.centos.src.rpm</a>   
-    </p>
-    </td>
-  </tr>
-</table>
-
-
-This procedure depicts an installation on version 6.5. If you are installing on
-7.X, substitute that package for your installation. 
+You use the same installation procedure for all versions of CentOS.
```
Comment (moxiegirl) [inline docs/installation/centos.md]: @jfrazelle  Thanks for taking an initial pass. 

This first sentence isn't actually true any more, it is only true if a user is using the installation script.  If the user is not using the script, the procedure requires different commands.   So, I've rewritten this one to illustrate.  The others need to use the same pattern.

## Install Docker Engine

There are two ways to install Docker Engine.  You can use `curl` with the  `get.docker.com` site. This method runs an installation script which installs via the `yum` package manager. Or you can install with the `yum` package manager directly yourself.

### Install with the script

You use the same installation procedure for all versions of CentOS.

1. Log into your machine as a user with `sudo` or `root` privileges.

2. Make sure your existing yum packages are up-to-date.

		$ sudo yum update
		
3. Run the Docker installation script.
		
		$ curl -sSL https://get.docker.com/ | sh

4. Start the Docker daemon.

		$ sudo service docker start

5. Verify `docker` is installed correctly by running a test image in a container.

		$ sudo docker run hello-world
		Unable to find image 'hello-world:latest' locally
		latest: Pulling from hello-world
		a8219747be10: Pull complete 
		91c95931e552: Already exists 
		hello-world:latest: The image you are pulling has been verified. Important: image verification is a tech preview feature and should not be relied on to provide security.
		Digest: sha256:aa03e5d0d5553b4c3473e89c8619cf79df368babd1.7.1cf5daeb82aab55838d
		Status: Downloaded newer image for hello-world:latest
		Hello from Docker.
		This message shows that your installation appears to be working correctly.

		To generate this message, Docker took the following steps:
		 1. The Docker client contacted the Docker daemon.
		 2. The Docker daemon pulled the "hello-world" image from the Docker Hub.
				(Assuming it was not already locally available.)
		 3. The Docker daemon created a new container from that image which runs the
				executable that produces the output you are currently reading.
		 4. The Docker daemon streamed that output to the Docker client, which sent it
				to your terminal.

		To try something more ambitious, you can run an Ubuntu container with:
		 $ docker run -it ubuntu bash

		For more examples and ideas, visit:
		 http://docs.docker.com/userguide/
		 
## Install without the script

1. Log into your machine as a user with `sudo` or `root` privileges.

2. Make sure your existing yum packages are up-to-date.

		$ sudo yum update
    
3. Add the yum repo yourself.

    For CentOS 6 run:

		$ cat >/etc/yum.repos.d/docker.repo <<-EOF
        [dockerrepo]
        name=Docker Repository
        baseurl=https://yum.dockerproject.org/repo/main/centos/6
        enabled=1
        gpgcheck=1
        gpgkey=https://yum.dockerproject.org/gpg
        EOF

    For CentOS 7 run:

		$ cat >/etc/yum.repos.d/docker.repo <<-EOF
        [dockerrepo]
        name=Docker Repository
        baseurl=https://yum.dockerproject.org/repo/main/centos/7
        enabled=1
        gpgcheck=1
        gpgkey=https://yum.dockerproject.org/gpg
        EOF

4. Install the Docker package.

        $ yum install docker-engine
        
4. Start the Docker daemon.

		$ sudo service docker start

5. Verify `docker` is installed correctly by running a test image in a container.

		$ sudo docker run hello-world

**Comment by jfrazelle** [comment]
moved the release script changes to https://github.com/docker/docker/pull/14745 because if we decide to push 1.8.0 to the old repo and new we will not want to merge the changes to the release.sh script. But this should be good to go I have tested it in all the containers we build the debs/rpms in

**Comment by jfrazelle** [comment]
ping @tianon

**Comment by jfrazelle** [comment]
ok i have tested a bajillion times i am confident :innocent: fingers crossed

File: hack/install.sh [hunk 4]
```diff
@@ -95,45 +100,60 @@ do_install() {
...
-			else
-				(
-					set -x
-					$sh_c 'sleep 3; yum -y -q install docker-io'
-				)
-			fi
+		amzn)
+			(
+			set -x
+			$sh_c 'sleep 3; yum -y -q install docker'
```
Comment (calavera) [inline hack/install.sh]: @jfrazelle just double checking that the name of the package here is still `docker` and not `docker-engine`.

File: hack/install.sh [hunk 5]
```diff
@@ -142,20 +162,20 @@ do_install() {
 
 		'opensuse project'|opensuse|'suse linux'|sled)
 			(
-				set -x
-				$sh_c 'sleep 3; zypper -n install docker'
+			set -x
+			$sh_c 'sleep 3; zypper -n install docker'
```
Comment (calavera) [inline hack/install.sh]: is this still `docker` and not `docker-engine`, right?

File: hack/install.sh [hunk 6]
```diff
@@ -33,7 +38,7 @@ echo_docker_as_nonroot() {
 	If you would like to use Docker as a non-root user, you should now consider
 	adding your user to the "docker" group with something like:
 
-	  sudo usermod -aG docker $your_user
+	sudo usermod -aG docker $your_user
```
Comment (tianon) [inline hack/install.sh]: Why this change?

File: hack/install.sh [hunk 7]
```diff
@@ -55,16 +60,16 @@ do_install() {
...
+		why we're displaying this warning and provide the opportunity to cancel the
+		installation.
 
-			If you installed the current Docker package using this script and are using it
-			again to update Docker, you can safely ignore this message.
+		If you installed the current Docker package using this script and are using it
+		again to update Docker, you can safely ignore this message.
 
-			You may press Ctrl+C now to abort this script.
+		You may press Ctrl+C now to abort this script.
```
Comment (tianon) [inline hack/install.sh]: Just making sure all this whitespace changing is intentional.  This one has no effect on the output (unlike the previous one I commented on), so I'm indifferent unless it was unintentional.

File: hack/install.sh [hunk 8]
```diff
@@ -95,45 +100,60 @@ do_install() {
...
+		dist_version="$(. /etc/lsb-release && echo "$DISTRIB_CODENAME")"
 	fi
 	if [ -z "$lsb_dist" ] && [ -r /etc/debian_version ]; then
 		lsb_dist='debian'
+		dist_version="$(cat /etc/debian_version | sed 's/\/.*//' | sed 's/\..*//')"
+		if [ "$dist_version" = "8" ]; then
+			dist_version="jessie"
+		elif [ "$dist_version" = "7" ]; then
+			dist_version="wheezy"
+		fi
```
Comment (tianon) [inline hack/install.sh]: Would this part be safer as a `case` statement so future additions are simpler?

File: hack/install.sh [hunk 9]
```diff
@@ -95,45 +100,60 @@ do_install() {
...
+			(
+			set -x
+			$sh_c 'sleep 3; yum -y -q install docker'
+			)
 			if command_exists docker && [ -e /var/run/docker.sock ]; then
 				(
-					set -x
-					$sh_c 'docker version'
+				set -x
+				$sh_c 'docker version'
```
Comment (tianon) [inline hack/install.sh]: More seemingly unintentional whitespace changes. :smile:

File: hack/install.sh (hunk 4 above)
Comment (tianon) [inline hack/install.sh]: @calavera yeah, I believe so since we aren't creating RPMs for Amazon Linux

File: hack/install.sh [hunk 10]
```diff
@@ -209,30 +229,51 @@ do_install() {
...
 				) || true
 			fi
 			echo_docker_as_nonroot
 			exit 0
 			;;
 
+		fedora|centos)
+			cat >/etc/yum.repos.d/docker${repo}.repo <<-EOF
+			[docker${repo}repo]
```
Comment (tianon) [inline hack/install.sh]: Can this be hyphen-separated or something so it doesn't become a gnarly-to-read run-on (`dockermainrepo` vs `docker-main-repo`)?

File: hack/install.sh [hunk 11]
```diff
@@ -209,30 +229,51 @@ do_install() {
...
+				set -x
+				$sh_c 'sleep 3; yum -y -q install docker-engine'
+				)
+			fi
+			if command_exists docker && [ -e /var/run/docker.sock ]; then
+				(
+				set -x
+				$sh_c 'docker version'
+				) || true
+			fi
```
Comment (tianon) [inline hack/install.sh]: Since we're repeating this `docker version` block a couple times now, do you think we should maybe move it into `echo_docker_as_nonroot` instead?

**Comment by tianon** [comment]
A few minor nits (especially whitespace), but overall looks great! :+1:

File: hack/install.sh (hunk 4 above)
Comment (jfrazelle) [inline hack/install.sh]: Ya those are not our packages still

File: hack/install.sh (hunk 7 above)
Comment (jfrazelle) [inline hack/install.sh]: I must have gg=G it by accident

File: hack/install.sh (hunk 5 above)
Comment (jfrazelle) [inline hack/install.sh]: yup

File: hack/install.sh (hunk 11 above)
Comment (jfrazelle) [inline hack/install.sh]: good call

**Comment by jfrazelle** [comment]
fixed and updated!, effing `gg=G` habit

File: hack/install.sh [hunk 12]
```diff
@@ -209,30 +227,39 @@ do_install() {
...
+			$sh_c "echo deb https://apt.dockerproject.org/repo ${lsb_dist}-${dist_version} ${repo} > /etc/apt/sources.list.d/docker.list"
+			$sh_c 'sleep 3; apt-get update; apt-get install -y -q docker-engine'
 			)
-			if command_exists docker && [ -e /var/run/docker.sock ]; then
+			echo_docker_as_nonroot
+			exit 0
+			;;
+
+		fedora|centos)
+			cat >/etc/yum.repos.d/docker${repo}.repo <<-EOF
```
Comment (tianon) [inline hack/install.sh]: `docker-${repo}.conf` ? :innocent:

Comment (tianon) [inline hack/install.sh]: Err `.repo`

**Comment by tianon** [comment]
Sorry noticed one more minor one /me hides

**Comment by jfrazelle** [comment]
hahaha no worries

**Comment by jfrazelle** [comment]
fixed!

**Comment by jfrazelle** [comment]
im re-running all my tests :)

**Comment by tianon** [comment]
Nice, LGTM :+1:

**Comment by jfrazelle** [comment]
ping @moxiegirl for docs :innocent:

**Comment by moxiegirl** [comment]
checking ...

**Comment by jfrazelle** [comment]
ping @thaJeztah for docs review and merge

File: docs/installation/centos.md [hunk 13]
```diff
@@ -112,6 +89,50 @@ Installation procedure:
...
+
+1. Log into your machine as a user with `sudo` or `root` privileges.
+
+2. Make sure your existing yum packages are up-to-date.
+
+		$ sudo yum update
+    
+3. Add the yum repo yourself.
+
+    For CentOS 6 run:
```
Comment (thaJeztah) [inline docs/installation/centos.md]: Erm, looks like CentOS 6 got back somehow here

File: docs/installation/centos.md [hunk 14]
```diff
@@ -112,6 +89,50 @@ Installation procedure:
...
 		For more examples and ideas, visit:
 		 http://docs.docker.com/userguide/
+
+### Install without the script
+
+1. Log into your machine as a user with `sudo` or `root` privileges.
+
+2. Make sure your existing yum packages are up-to-date.
+
+		$ sudo yum update
```
Comment (thaJeztah) [inline docs/installation/centos.md]: s/tabs/spaces/ :wink:

File: docs/installation/centos.md (hunk 13 above)
Comment (calavera) [inline docs/installation/centos.md]: :scream_cat:

File: docs/installation/centos.md [hunk 15]
```diff
@@ -112,6 +89,50 @@ Installation procedure:
...
+        gpgkey=https://yum.dockerproject.org/gpg
+        EOF
+
+4. Install the Docker package.
+
+        $ sudo yum install docker-engine
+        
+5. Start the Docker daemon.
+
+		$ sudo service docker start
```
Comment (thaJeztah) [inline docs/installation/centos.md]: tabs?

File: docs/installation/centos.md [hunk 16]
```diff
@@ -112,6 +89,50 @@ Installation procedure:
...
+
+6. Verify `docker` is installed correctly by running a test image in a container.
+
+		$ sudo docker run hello-world
```
Comment (thaJeztah) [inline docs/installation/centos.md]: here as well (tabs?)

**Comment by thaJeztah** [comment]
Thanks, @jfrazelle! See inline comments above. I think your changes contain more tabs instead of spaces, but probably easier for you to check (it's no real show-stopper, we can fix that up later if you want)

**Comment by thaJeztah** [comment]
Oh, apart from the above, LGTM. Very nice!

**Comment by jfrazelle** [comment]
fixed!

**Comment by thaJeztah** [comment]
Thanks!
